core/include/scipp/core/element/arithmetic.h

```diff
@@ -22,19 +22,26 @@ constexpr auto add_inplace_types =
              std::tuple<float, int64_t>, std::tuple<float, int32_t>,
              std::tuple<int64_t, bool>>;
 
-constexpr auto add_inplace_nan_types =
-    arg_list<double, float, std::tuple<double, float>,
-             std::tuple<float, double>>;
+template <class T> struct ValueType { using value_type = T; };
+template <class T> struct ValueType<ValueAndVariance<T>> {
+  using value_type = T;
+};
 
 constexpr auto plus_equals =
     overloaded{add_inplace_types, [](auto &&a, const auto &b) { a += b; }};
 constexpr auto nan_plus_equals =
-    overloaded{add_inplace_nan_types, [](auto &&a, const auto &b) {
-                 using std::isnan;
-                 if (isnan(a))
-                   a = std::decay_t<decltype(a)>{0}; // Force zero
-                 if (!isnan(b))
+    overloaded{add_inplace_types, [](auto &&a, const auto &b) {
```

Comment:
I think you need to check that both a and b are floating point types for this overload as isnan is called on both of them. Alternatively could remove `std::tuple<double, int64_t>, std::tuple<double, int32_t>,
             std::tuple<float, int64_t>, std::tuple<float, int32_t>,` from the allowed input types.

Reply:
It may turn out to wrap std:isnan to Integral types on windows in this case.

Reply:
> It may turn out to be easier to wrap std:isnan to handle Integral types on windows in this case.

Yes, woulnd't that simplify quite a few things?

Reply:
> Yes, woulnd't that simplify quite a few things? There is not even need for wrapping. Just define isnan(ints) in the namespace where the kernel is defined, and it should be found with ADL?

It would also need to be used internally with the overload in `ValueAndVariance`. - But it's a good suggestion to maybe have our own special functions at the basic type level which we use directly instead of those from `std::`